python/ray/serve/http_state.py

```diff
@@ -54,9 +55,24 @@ def _get_target_nodes(self) -> List[Tuple[str, str]]:
 
         if location == DeploymentMode.HeadOnly:
             head_node_resource_key = get_current_node_resource_key()
-            target_nodes = [(node_id, node_resource)
-                            for node_id, node_resource in target_nodes
-                            if node_resource == head_node_resource_key][:1]
+            return [(node_id, node_resource)
+                    for node_id, node_resource in target_nodes
+                    if node_resource == head_node_resource_key][:1]
+
+        if location == DeploymentMode.FixedNumber:
+            num_replicas = self._config.fixed_number_replicas
+            if num_replicas > len(target_nodes):
+                logger.warning(
+                    "You specified fixed_number_replicas="
+                    f"{num_replicas} but there are only "
+                    f"{len(target_nodes)} total nodes. Serve will start one "
+                    "HTTP proxy per node.")
+                num_replicas = len(target_nodes)
+
+            # Seed the random state so sample is deterministic.
+            # i.e. it will always return the same set of nodes.
+            random.seed(0)
```

Comment:
Can you please add a unit test for this behavior? e.g., pull the node-selecting logic out into a utility and add some tests for it.

Reply:
Done.